sunpy/net/helioviewer.py

```diff
@@ -101,20 +100,13 @@ def download_jp2(self, date, directory=None, overwrite=False, **kwargs):
         ----------
         date : `datetime.datetime`, string
             A string or datetime object for the desired date of the image
-        directory : string
-            (Optional) Directory to download JPEG 2000 image to.
-        observatory : string
-            (Optional) Observatory name
-        instrument : string
-            (Optional) instrument name
-        detector : string
-            (Optional) detector name
-        measurement : string
-            (Optional) measurement name
         sourceId : int
-            (Optional) data source id
```

Comment:
Why did you remove these?

Reply:
Because when I used those parameters the map function gave me an error saying that the file format is not supported by sunpy

Reply:
They are passed into Helioviewer Function but these lines below throw an error 
>>> filepath = hv.download_jp2('2012/07/03 14:30:00', observatory='SDO', instrument='AIA', detector='AIA', measurement='171')   
>>> aia = sunpy.map.Map(filepath)

Reply:
Doesn't that mean that the return from the function was broken? 
You should check that the file returned is the same before and after this change. 
I am not in favour of changing the function call without a reason.

Reply:
Oh ok I will have a look at it again

Reply:
Ok sure

Reply:
I have to change function call a bit cause till now sourceID was optional but now its necessary

Reply:
Ok, so updating to v2 API means we have to change the function? Then sure.

Reply:
Yes according to me we need to change the way we pass the parameters because passing the sourceId is compulsory in version 2 of the API

Reply:
Ok, then those changes need to be made. This PR still needs a rebase as well.

Reply:
Yup I will try doing that once I get the code working
Actually, I have just started Git so I am not that comfortable with all the commands but I will try

Reply:
Yeah git is a large pain. It is incredibly easy to make a mistake. I do all the the time.

Reply:
Haha I will learn how to rebase on the way